doc/storage/api/api.rst

```diff
@@ -63,7 +63,19 @@ These definitions must be defined in the header file :file:`psa/storage_common.h
     .. field:: psa_storage_create_flags_t flags
         The flags set when the ``uid`` was create
 
+.. struct:: psa_storage_iterator_t
 
+    .. summary::
+        An implantation defined opaque structure containing the context for an iterator. 
+        The structure MUST  contain all all the state required by the iterator. 
+        No further state is retained by the implementation. 
+        
+        The structure is initilaised by the `ps_iterator_start` function. 
+        It is modified by the `ps_iterator_next` function. 
+        
+        It is the duty of the calling application to free memory when the iterator is no longer required. 
```

Comment:
As there is no hidden allocation, using the term 'free' might be confusing. Might it be better to state that the application can discard or reuse the iterator object once it has finished using it. This can be before, or after, the iterator has reached the end of the iteration.